targets/unity-v2/make.js

```diff
@@ -378,24 +417,32 @@ function getAuthParams(apiCall) {
     return "AuthType.None";
 }
 
-function getRequestActions(tabbing, apiCall) {
-    if (apiCall.name === "GetEntityToken")
+function getRequestActions(tabbing, apiCall, isApiInstance = false) {
+    if (apiCall.name === "GetEntityToken" && isApiInstance === false)
         return tabbing + "AuthType authType = AuthType.None;\n" +
             "#if !DISABLE_PLAYFABCLIENT_API\n" +
-            tabbing + "if (authType == AuthType.None && PlayFabClientAPI.IsClientLoggedIn())\n" +
+            tabbing + "if (authType == AuthType.None && (request.AuthenticationContext != null ? request.AuthenticationContext.IsClientLoggedIn() : PlayFabClientAPI.IsClientLoggedIn()))\n" +
             tabbing + "    authType = AuthType.LoginSession;\n" +
             "#endif\n" +
             "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API || UNITY_EDITOR\n" +
-            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(PlayFabSettings.DeveloperSecretKey))\n" +
+            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(request.AuthenticationContext.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey))\n" +
             tabbing + "    authType = AuthType.DevSecretKey;\n" +
             "#endif\n";
+	if (apiCall.name === "GetEntityToken" && isApiInstance === true)
+        return tabbing + "AuthType authType = AuthType.None;\n" +
+            "#if !DISABLE_PLAYFABCLIENT_API\n" +
+            tabbing + "if (authType == AuthType.None && (request.AuthenticationContext != null ? request.AuthenticationContext.IsClientLoggedIn() : (authenticationContext != null ? authenticationContext.IsClientLoggedIn() : PlayFabClientAPI.IsClientLoggedIn())))\n" +
+            tabbing + "    authType = AuthType.LoginSession;\n" +
+            "#endif\n" +
+            "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API || UNITY_EDITOR\n" +
+            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(request.AuthenticationContext.DeveloperSecretKey ?? (authenticationContext.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey)))\n" +
+            tabbing + "    authType = AuthType.DevSecretKey;\n" +
+            "#endif\n";		
 
     if (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest")
         return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n";
     if (apiCall.auth === "SessionTicket")
         return tabbing + "if (!IsClientLoggedIn()) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn,\"Must be logged in to call this method\");\n";
```

Comment:
>if (!IsClientLoggedIn()) [](start = 26, length = 24)

Replace 

if (!IsClientLoggedIn())

with two different cases:

when isApiInstance === true:

if (string.IsNullOrEmpty(request.AuthenticationContext.ClientSessionTicket ?? (authenticationContext.ClientSessionTicket ?? PluginManager.GetPlugin<IPlayFabTransportPlugin>(PluginContract.PlayFab_Transport).AuthKey)))

when isApiInstance === false:

if (string.IsNullOrEmpty(request.AuthenticationContext.ClientSessionTicket ?? PluginManager.GetPlugin<IPlayFabTransportPlugin>(PluginContract.PlayFab_Transport).AuthKey))

Reply:
isApiInstance === false case got null pointer exception when ReportDeviceInfo called so I keep instanceapi.ejs.cs and isApiInstance  === false case as :"if (!IsClientLoggedIn()) . Committed last version